spec/rubocop/cop/style/double_pipe_equals_spec.rb

```diff
@@ -0,0 +1,94 @@
+# frozen_string_literal: true
+
+describe RuboCop::Cop::Style::DoublePipeEquals do
+  let(:config) { RuboCop::Config.new }
+  subject(:cop) { described_class.new(config) }
+
+  context 'when using var = var ? var : something' do
+    it 'registers an offense' do
+      expect_offense(<<-RUBY.strip_indent)
+        foo = foo ? foo : 'default'
+        ^^^^^^^^^^^^^^^^^^^^^^^^^^^ Use the double pipe equals operator `||=` instead.
+      RUBY
+
+      expect_offense(<<-RUBY.strip_indent)
+        foo = if foo
+        ^^^^^^^^^^^^ Use the double pipe equals operator `||=` instead.
+                foo
+              else
+                'default'
+              end
+      RUBY
+    end
+
+    it 'autocorrects to `var ||= something`' do
+      new_source = autocorrect_source('foo = foo ? foo : 3')
+      expect(new_source).to eq('foo ||= 3')
+
+      new_source2 = autocorrect_source(<<-RUBY.strip_indent)
+        foo = if foo
+                foo
+              else
+                3
+              end
+      RUBY
+      expect(new_source2).to eq('foo ||= 3')
```

Comment:
Because a heredoc includes a new line at end.

```
[1] pry(main)> <<-END == "foo\n"
[1] pry(main)* foo
[1] pry(main)* END
=> true
[2] pry(main)> 
```